server/core/src/mapper.ts

```diff
@@ -66,21 +72,32 @@ function findSearchPresenter (hierarchy: Hierarchy, _class: Ref<Class<Doc>>): Se
 /**
  * @public
  */
-export async function updateDocWithPresenter (hierarchy: Hierarchy, doc: IndexedDoc): Promise<void> {
-  const searchPresenter = findSearchPresenter(hierarchy, doc._class)
+export async function updateDocWithPresenter (
+  hierarchy: Hierarchy,
+  doc: DocIndexState,
+  elasticDoc: IndexedDoc,
+  refDocs: {
+    parentDoc: DocIndexState | undefined
+    spaceDoc: DocIndexState | undefined
+  }
+): Promise<void> {
+  const searchPresenter = findSearchPresenter(hierarchy, elasticDoc._class)
   if (searchPresenter === undefined) {
     return
   }
 
-  const reader = createIndexedReader(doc._class, hierarchy, doc)
+  const reader = createIndexedReader(elasticDoc._class, hierarchy, doc, {
```

Comment:
As I remember we discussed to get rid of reateIndexedReader and use Elastic DocIndexState attributes?

Reply:
Yes. I kept it, but now it uses DocIndexState from mongo. I didn't want to split get and getDoc into two functions and kept it in one

Reply:
I do not see changes related to it.